pipenv/cli/command.py

```diff
@@ -709,5 +709,24 @@ def verify(state):
     sys.exit(0)
 
 
+@cli.command(
+    short_help="Generate a requirements.txt from Pipfile.lock.",
+    context_settings=CONTEXT_SETTINGS,
+)
+@option("--dev", is_flag=True, default=False, help="Also add development requirements.")
+@pass_state
+def requirements(state, dev=False):
+    lockfile = state.project.lockfile_content
+    for i, package_index in enumerate(lockfile['_meta']['sources']):
+        prefix = '-i' if i == 0 else '--extra-index-url'
+        echo(crayons.normal(' '.join([prefix, package_index['url']])))
```

Comment:
This should output the index verbatim as specified in the Pipfile, _without_ expanding variables. It is fairly common to specify credentials for a private PyPI this way, and writing those into the resulting requirements.txt file is counter-productive.

I'm not sure whether your existing code is already doing this or not, just wanted to mention this need.

Reply:
@rittneje interesting point. I did a couple of tests and for me the $ENV_VAR gets substituted in the output and ${ENV_VAR} does not. I am currently just echoing the value from Pipfile.lock to stdout. Do you have any suggestions how to prevent this substitution in a smart way? I am still thinking about it